### ISSUE TITLE
Font management

### ISSUE THREAD

**Issue by tangledhelix** [DistributedProofreaders/guiguts-py#66]
Choose a default font (for each platform Win/Mac/Linux):

- should ship with the platform by default
- should be monospace / fixed-width

I would also propose bundling DP Sans Mono for easy use, if possible. I'm not sure how to install custom fonts on every platform, but on macOS I think adding font files to `$HOME/Library/Fonts/` is sufficient.

"Linux" is actually a bunch of different platforms, really... I'll leave it for someone else to figure out what default fonts are available on any of them, since I have no clue about it...

**Comment by srjfoo** [comment]
Modern Macs should all come with **Font Book**, and all you should need to do is open that and navigate to where the downloaded font is and open it.

Caveat: The current version of DPSM that I have is, I think (though I haven't verified it for sure), the current version, and does not identify itself to at least some apps (Font Book and BBEdit, for sure) as a monospace font, though it behaves like one. (In Font Book, it sorts into All and User fonts for me.)

**Comment by tangledhelix** [comment]
That's true (about Font Book) but I was hoping to auto-install it, and reduce the amount of extra steps for users to take. All Font Book really does is provide a UI for this - I believe dropping font files into the right folder accomplishes the same thing.

(What I'm not sure about is whether filesystem sandboxing would prevent an install from happening.)

Regarding your caveat: I notice that if I look in my Font book, it has a filter for fixed-width fonts, and choosing it, DP Sans Mono is not there.

<img width="745" alt="Screenshot 2023-12-19 at 1 30 34 AM" src="https://github.com/DistributedProofreaders/guiguts-py/assets/294481/918047e0-5062-4ea0-a37f-4b7a985f6753">

That filter checks that the "Design Style" field is set to "Monospaced". So it seems like that bit of metadata may not be set quite right in the DP Sans Mono font... too bad. Although it doesn't prevent it from working in Guiguts.

**Comment by tangledhelix** [comment]
Also, for what it's worth, it seems like bundling font files in apps may be common and we could do the same?

```
❯ find /Applications -type f -name '*.ttf' | wc -l
    1123
```

**Comment by srjfoo** [comment]
> That's true (about Font Book) but I was hoping to auto-install it, and reduce the amount of extra steps for users to take. All Font Book really does is provide a UI for this - I believe dropping font files into the right folder accomplishes the same thing.
> 
> (What I'm not sure about is whether filesystem sandboxing would prevent an install from happening.)
> 
> Regarding your caveat: I notice that if I look in my Font book, it has a filter for fixed-width fonts, and choosing it, DP Sans Mono is not there.
> <img alt="Screenshot 2023-12-19 at 1 30 34 AM" width="745" src="https://private-user-images.githubusercontent.com/294481/291496397-918047e0-5062-4ea0-a37f-4b7a985f6753.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NzIxMjEsIm5iZiI6MTcwMjk3MTgyMSwicGF0aCI6Ii8yOTQ0ODEvMjkxNDk2Mzk3LTkxODA0N2UwLTUwNjItNGVhMC1hMzdmLTRiN2E5ODVmNjc1My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxOVQwNzQzNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYWQzZDE2NmNlZjBjYjBkZjIyYzdmNzUyZTkwODFjZjFiNjJiMTc3ZGQ3NmI0ODFhYTE2MzdmYmFkYWE0NWM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.mt2IhC6V8kwnxp8-Ikl1hF3Wy3h6tIQpexT_96RRFgE">
> 
> That filter checks that the "Design Style" field is set to "Monospaced". So it seems like that bit of metadata may not be set quite right in the DP Sans Mono font... too bad. Although it doesn't prevent it from working in Guiguts.

Okay, that's fair. I missed the fact that you were thinking of an auto-install (I figured it would be easier for users to use Font Book than to manually move the font to the correct directlry.

I didn't know what exactly it was that caused DPSM not to be identified as monospaces, but figured it was some sort of metadata that was not set properly. And yes, same. In Font Book, DPSM is not identified as monospaced for me, either.

**Comment by cpeel** [comment]
I agree that having a sensible OS-specific default font is a great idea.

I don't think GG should be _installing_ fonts on a user's system -- I've never seen any other app do that on any operating system. The more we can make GG act like "any other app" the easier it will be for users to know how to install / uninstall it.

I've no problem with bundling fonts and/or providing the user instructions for how to install them.

> Also, for what it's worth, it seems like bundling font files in apps may be common and we could do the same?

Looks like [ATSApplicationFontsPath](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/GeneralPurposeKeys.html#//apple_ref/doc/uid/TP40009253-SW8) is the macOS way to have apps use fonts included in their bundle.

**Comment by windymilla** [comment]
Default fonts are now set (choosing DPSM if available, or DejaVu Sans Mono, or Courier New, or Courier). Apparently Courier is guaranteed to be available when using Tk.
I'm therefore punting the rest of this to "future feature" (probably meaning "after the first beta release")
